### DISCUSSION TITLE
Ability to configure certain fields on Logbook and Dashboard pages

### DISCUSSION THREAD

**Discussion by timkirby** [magicbug/Cloudlog#952] [Ideas / Feature Requests]
As a predominantly VHF operator, it would be very helpful to me to give Locator square a bit more prominence.

In a perfect world, I'd like the ability to display it on the Dashboard view, but recognise that perhaps there isn't so much space available here. However, on the Logbook view, I'd lose the Country field in a heartbeat (I know what country 2M0SQL represents!) and replace it with Locator. 

That clearly wouldn't work for a mostly HF operator - but could it be possible to select which fields are displayed on that screen - Country/Locator/SOTA/IOTA etc? Could that be defined as part of a Station Profile or similar?

Thanks as ever, to all of you who develop the features on Cloudlog. Your work and innovation is much appreciated here.

Tim GW4VXE/GW4MM

**Comment by magicbug** [comment]
Somewhat a tricky one, I'll add to the road map but not sure when.

Think user account is the best place for this and just have a few Columns where you can select the output from a drop-down.

**Comment by AndreasK79** [comment]
Shouldn't be too hard to implement with a dropdown to choose what goes where. Then a switch in the view to determine which value to put out. 
I guess a version of that is what @magicbug was thinking about?

**Reply by magicbug** [reply]
@AndreasK79 Yes think thats probably the easiest way to implement this have options for the Dashboard and Logbook page matching the column counts as they stand but allowing people to change them.

**Comment by timkirby** [comment]
Thanks very much both!

I guess if the fully flexible approach was too much of a pain to implement, then a limited number of predefined views for the Logbook screen (eg VHF, DXCC, IOTA, SOTA) that you could select in your profile/station location would perhaps do the trick?

**Comment by AndreasK79** [comment]
It's getting there:
![image](https://user-images.githubusercontent.com/6977712/117124954-3f633580-ad99-11eb-926c-ecdc0d6fc87b.png)
![image](https://user-images.githubusercontent.com/6977712/117125062-5dc93100-ad99-11eb-8c00-cd0f111aaf50.png)
![image](https://user-images.githubusercontent.com/6977712/117125137-71749780-ad99-11eb-8d3b-44fb12fa1959.png)
I've made the last 4 columns user definable for the moment.

@magicbug & @timkirby does this seem like an ok solution? Input is welcome.

**Comment by timkirby** [comment]
Hi Andreas

Thank you very much for your work on this! It looks like you have solved it nicely and made it very flexible to cater for a variety of types of operation and interests.

Tim

**Reply by AndreasK79** [reply]
Thanks Tim. Right now it's only working on the dashboard. I need to make it work in the logbook as well. I also need to add more choices in the dropdown.

**Comment by timkirby** [comment]
Thanks Andreas! That’s really good news. 

Tim

**Comment by AndreasK79** [comment]
Tweaked the layout for the selection in the user account settings:
![image](https://user-images.githubusercontent.com/6977712/117149329-6af51880-adb7-11eb-95df-6ec008f5c264.png)

**Comment by AndreasK79** [comment]
#1019 Added. @magicbug please try it out and give me some feedback. The only thing that should be added, are translations. I haven't looked too much into that yet.

**Comment by timkirby** [comment]
Thank you!!

**Comment by magicbug** [comment]
@AndreasK79 looks good from the screenshots, sorry I've not had time to look today its been a bit of a struggle

**Reply by AndreasK79** [reply]
Translations has been added now.

**Comment by magicbug** [comment]
This looks great @AndreasK79 I've merged it as its currently stable and allows some further feedback.

@timkirby I've updated your install so you should find it under your account settings :)

**Reply by AndreasK79** [reply]
@magicbug I didn't make it fully configurable. Perhaps I should have done that, but the biggest job is done. It's easier to expand it now.

**Reply by magicbug** [reply]
I think it will keep a lot of people happy as is, however I'm sure there will be some feedback

**Comment by timkirby** [comment]
Fantastic! Going to play now!

Thank you both very much. 

Tim

**Comment by timkirby** [comment]
It’s working well and I can now put locator on both the logbook and dashboard screens which I am very excited about!

My only ‘wouldn’t it be nice’ is whether it would be possible to include the ‘Country’ field (ie the first 5 fields rather than 4) in the configurable fields. I think the other fields make good sense to be ‘fixed’

But thank you so much, Andreas - this is very useful to me.

Tim

**Reply by AndreasK79** [reply]
@timkirby Yeah, I knew the Country in logbook would be a "problem" when I made this. I've now added a fifth column, only used in logbook.
![image](https://user-images.githubusercontent.com/6977712/117404878-05b23c00-af0b-11eb-8f7e-6655d5c96c97.png)


@magicbug The PR also contains a bugfix for adding user accounts. The new columns were never set when creating a new user.

**Comment by timkirby** [comment]
Thanks Andreas! Very much appreciated!

Tim

**Comment by magicbug** [comment]
Merged so will be in hosted tomorrow morning for @timkirby

**Comment by timkirby** [comment]
Thanks Pete!

**Comment by timkirby** [comment]
Morning! Thanks for merging the changes in. There’s a slight issue on the logbook screen.

Although the column heading reflects the selected data (in my case ‘Gridsquare’) the rows from the logbook are still showing country rather than grid square for the ‘5th column’. Columns 1-4 are working as desired.

Sorry to be the bearer of bad news but thank you both for all your help.

Tim

**Reply by AndreasK79** [reply]
No worries Tim. I see the problem. I'm sure I did fix this, but somehow it disappeared when comitting the code. I'll push a fix for it.

**Reply by AndreasK79** [reply]
#1024 pushed. Now Peter needs to merge it.

**Reply by magicbug** [reply]
Merged I'll update Tims install in a bit :)

**Comment by timkirby** [comment]
Thanks Andreas! Peter kindly updated my install and it’s now looking great. Thank you for all your work on this.
